projects/transformers/integrations.py

```diff
@@ -0,0 +1,54 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2021, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+import math
+
+import wandb
+from transformers.integrations import INTEGRATION_TO_CALLBACK, WandbCallback
+
+
+"""
+This file serves to extend the functionalities of automated integrations such as those
+for wandb.
+"""
+
+
+class CustomWandbCallback(WandbCallback):
```

Comment:
Given this callback replaces the original one in wandb, for any metrics we want to track we should modify this class?

Reply:
Potentially. Some metrics may be better suited for a different callback. I think it will depend. I see the `CustomWandbCallback` being suitable for a metric that applies to all experiments. On the other hand, something that tracks sparsity can log to wandb directly by calling `wandb.log`.